osquery/core/query.cpp

```diff
@@ -72,17 +86,21 @@ bool Query::isNewQuery() const {
   return (query != query_.query);
 }
 
-Status Query::addNewResults(const QueryData& qd, const uint64_t epoch) const {
+Status Query::addNewResults(const QueryData& qd,
+                            const uint64_t epoch,
+                            uint64_t* counter) const {
```

Comment:
Why the use of pointers throughout? Would a `uint64_t counter` not suffice?

Reply:
Too much C :) These should be uint64_t&
Counter value is returned to the caller. Fixed.